packages/react-devtools-shared/src/backend/views/TraceUpdates/index.js

```diff
@@ -86,6 +96,11 @@ function traceUpdates(nodes: Set<HostInstance>): void {
       rect = measureNode(node);
     }
 
+    let displayName = this.showNames && agent.getComponentNameForHostInstance(node);
+    if (displayName != null && displayName.startsWith('Forget(')) {
+      displayName = '✨' + displayName.slice(7, -1);
+    }
```

Comment:
There is a chance you might end up receiving `TypeError`: calling `displayName.startsWith(...)` if `this.showNames` is `false`.

Reply:
Thank for pointing that out. I’ve updated the code to avoid the potential TypeError by ensuring `displayName` is only processed when `showNames` is true.

I also revamped the display name parsing logic to make it more extensible. As part of this, I added a 🧠 icon for components wrapped in `React.Memo`, and display names are now consistently unwrapped from HOCs for better clarity. Let me know if there’s anything else!